osu.Game/Overlays/Rankings/Displays/RankingsDisplay.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Bindables;
+using osu.Game.Rulesets;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Allocation;
+using System.Threading;
+using System;
+using osu.Game.Graphics;
+using osuTK.Graphics;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Graphics.Shapes;
+
+namespace osu.Game.Overlays.Rankings.Displays
+{
+    public abstract class RankingsDisplay<T> : OverlayView<T>, IHasCurrentValue<RulesetInfo>
+        where T : class
+    {
+        public Action StartLoading;
+        public Action FinishLoading;
+
+        private readonly BindableWithCurrent<RulesetInfo> current = new BindableWithCurrent<RulesetInfo>();
+
+        public Bindable<RulesetInfo> Current
+        {
+            get => current.Current;
+            set => current.Current = value;
+        }
+
+        /// <summary>
+        /// Whether we should create content on request success.
+        /// </summary>
+        protected virtual bool CreateContentOnSucess => true;
```

Comment:
The reason for this is that spotlights display has 2 requests in it and first request is only to get available spotlights, so there's no content to add. But if we will still try to add `Empty()` - animation will look incorrect since if we already have a table - it will be removed at first and added another table later. With this wew are kinda blocking content addition on first request success, so there will be no flickering and one table will softly transit to another

Reply:
then why isn't `OnSuccess` overridden instead of adding this flag which complicates matters even more?

Reply:
wow, that's a nice point of view